taxi/views.py

```diff
@@ -57,12 +84,48 @@ class ManufacturerDeleteView(LoginRequiredMixin, generic.DeleteView):
 class CarListView(LoginRequiredMixin, generic.ListView):
     model = Car
     paginate_by = 5
-    queryset = Car.objects.select_related("manufacturer")
+
+    def get_context_data(self, *, object_list=None, **kwargs) -> dict:
+        context = super().get_context_data(**kwargs)
+        car_model = self.request.GET.get("model", "")
+        context["search_form"] = CarSearchForm(
+            initial={"model": car_model}
+        )
+
+        return context
+
+    def get_queryset(self) -> QuerySet:
+        queryset = Car.objects.select_related("manufacturer")
+        car_model = self.request.GET.get("model")
+        if car_model:
+            return queryset.filter(
+                model__icontains=car_model
+            )
+
+        return queryset
 
 
 class CarDetailView(LoginRequiredMixin, generic.DetailView):
     model = Car
 
+    def get_context_data(self, *, object_list=None, **kwargs) -> dict:
+        context = super().get_context_data(**kwargs)
+        context["search_form"] = DriverSearchForm(initial={
+            "username": self.request.GET.get("username", "")
+        })
+        return context
+
+    def get_queryset(self) -> QuerySet:
+        queryset = get_user_model().objects.all()
+        driver_username = self.request.GET.get("username")
```

Comment:
In the `CarDetailView`, the `get_queryset` method is querying the user model instead of the car model. This seems incorrect for a car detail view and should be revised to query car details.